apps/desktop-wallet/src/features/assetsLists/TokenBalancesRow.tsx

```diff
@@ -44,49 +41,52 @@ const TokenBalancesRow = ({ tokenId, addressHash, isExpanded }: TokenBalancesRow
   const { t } = useTranslation()
   const theme = useTheme()
   const fiatCurrency = useAppSelector((s) => s.settings.fiatCurrency)
-  const { balances, logoUri, info, worth, isUnlisted, isLoading } = useAddressesToken(tokenId, addressHash)
+  const { data: tokens, isLoadingFTs } = useAddressesDisplayTokens(addressHash)
+  const token = tokens.find((token) => token.id === tokenId)
 
-  if (!balances) return null
+  if (!token) return null
+
+  const { image, name, symbol, balance, availableBalance, decimals, worth } = getTokenDisplayData(token)
 
   return (
     <TableRow key={tokenId} role="row" tabIndex={isExpanded ? 0 : -1}>
       <TokenRow>
-        <AssetLogoStyled assetImageUrl={logoUri} size={30} assetName={info?.name} />
+        <AssetLogoStyled assetImageUrl={image} size={30} assetName={name} />
         <NameColumn>
           <TokenName>
-            {info?.name ?? <HashEllipsed hash={tokenId} tooltipText={t('Copy token hash')} disableCopy={!isExpanded} />}
-            {isUnlisted && (
+            {name ?? <HashEllipsed hash={tokenId} tooltipText={t('Copy token hash')} disableCopy={!isExpanded} />}
+            {token.type === 'unlistedFT' && (
```

Comment:
The summary of this discussion was that by using the `type` property (as I was inspired by your suggestion in the [`sendAnalytics` function](https://github.com/alephium/alephium-frontend/blob/e0056c7fecfd5275ab0734bb130f5803d230e507/apps/mobile-wallet/src/analytics.tsx#L39-L74) we have stricter components since we are always forced to check for its type before knowing what properties are available. The example I gave on the call was that in the past we sometimes forgot to pass the `decimals` when rendering an `Amount` (or `Badge`, I don't recall) component and since it was always optional (to accommodate rendering any type of token) the compiler never complained. After reworking some of them they now accept different props based on the type of token, so our code is more type-strict.

Reply:
Very clear! The thing that I don't understand is how using type guards would be less type-strict?
When you go 
```ts
isListedFT(token) && token.decimals
``` 
the compiler knows that `token` is of type `ListedFT` and has required `decimals`

Reply:
I can demonstrate in the video on Slack (too big to upload here)